### PR TITLE
feat: username to customerId migration endpoint

### PR DESCRIPTION
**Pull request by maciejkorsan**
I've added migration endpoint which copies username to customerId field (needed by customer-io to track user properly)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +55,7 @@ Install package in your project:
 ```sh
 cd my_project
 npm install @syncano/cli --save-dev
-npm i @eyedea-sockets/syncano-socket-customer-io --save
+npm install @eyedea-sockets/<name> --save
```
Comment (Idered) [inline README.md]: Why have you changed to `<name>`?

File: socket.yml [hunk 2]
```diff
@@ -67,6 +67,15 @@ endpoints:
...
           type: string
           description: Unique user id
+  migrate:
+    description: username to customerId migration
+    required:
+      - migrationKey
+    inputs:
+      properties:
+        migrationKey:
+          type: string
```
Comment (Idered) [inline socket.yml]: `required` property doesn't check if value is not empty, it only checks if key is present in request data. Add `minLength` for extra validation

File: src/migrate.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  migrationKey: string
+}
+
+class Endpoint extends S.Endpoint {
+  async run({response, users}: S.Core, {args, config}: S.Context<Args>) {
+    if (args.migrationKey === config.CUSTOMERIO_MIGRATION_KEY) {
+      try {
+        const allUsers = await users.list()
+        for (const user of allUsers) {
+          if (user.username && !user.customerId) {
```
Comment (Idered) [inline src/migrate.ts]: Syncano users can't have empty username so you can skip first check. I would move `!user.customerId` to `await users.whereNotNull('customerId').list()`

File: src/migrate.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            users
```
Comment (Idered) [inline src/migrate.ts]: You can replace this with mass update:

```
await data.book.update([
  14, {title: 'Book with id 14 was updated'},
  15, {title: 'Book with id 15 was updated'}
])
```

Comment (maciejkorsan) [reply src/migrate.ts]: It'll make code less readable

File: src/migrate.ts (hunk 3 above)
Comment (maciejkorsan) [reply src/migrate.ts]: whereNull? (I need all users that don't have a customerId, whereNotNull will return a list with users WITH customerId)

File: src/migrate.ts (hunk 4 above)
Comment (Idered) [reply src/migrate.ts]: Performance > Readability

Also it won't be that complicated. 

```jsx
const usernameToCustomerId = item => ([item.id, {customerId: item.username}])
await users.update(allUsers.map(usernameToCustomerId))
```

Comment (maciejkorsan) [reply src/migrate.ts]: but on the other hand it'll be a single database call - makes sense.

File: src/migrate.ts (hunk 3 above)
Comment (Idered) [reply src/migrate.ts]: Right, you need whereNull. My mistake

File: src/migrate.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+class Endpoint extends S.Endpoint {
+  async run({response, users}: S.Core, {args, config}: S.Context<Args>) {
+    if (args.migrationKey === config.CUSTOMERIO_MIGRATION_KEY) {
+      try {
+        const allUsers = await users.whereNull('customerId').list()
+        const updateList = []
+        for (const user of allUsers) {
+            updateList.push([
```
Comment (Idered) [inline src/migrate.ts]: Invalid indent :P

**Pull request merged**